mirdata/datasets/compmusic_otmm_makam.py

```diff
@@ -0,0 +1,249 @@
+        checksum="c2b9c8bdcbdcf15745b245adfc793145",
+        destination_dir=None,
+    )
+}
+
+LICENSE_INFO = (
+    "Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License"
+)
+
+
+DATA = core.LargeData("otmm_makam_index.json")
+
+
+class Track(core.Track):
+    """OTMM Makam Track class
+
+    Args:
+        track_id (str): track id of the track
+        data_home (str): Local path where the dataset is stored. default=None
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+
+    Attributes:
+        makam (str): string referring to the makam represented in the track
+        tonic (float): tonic annotation
+        mbid (str): MusicBrainz ID of the track
+
+    Cached Properties:
+        pitch (F0Data): pitch annotation
+        mb_tags (dict): dictionary containing the raw editorial track metadata from music brainz
+
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+        # Annotation paths
+        self.pitch_path = core.none_path_join(
+            [self._data_home, self._track_paths["pitch"][0]]
+        )
+        self.mb_tags_path = core.none_path_join(
+            [self._data_home, self._track_paths["metadata"][0]]
+        )
+
+        # Get attributes
+        self.tonic = self._track_metadata.get("tonic")
+        self.makam = self._track_metadata.get("makam")
+        self.mbid = self._track_metadata.get("mbid")
+
+    @core.cached_property
+    def pitch(self):
+        return load_pitch(self.pitch_path)
+
+    @core.cached_property
+    def mb_tags(self):
+        return load_mb_tags(self.mb_tags_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            f0_data=[(self.pitch, "pitch")],
+            metadata={
+                "makam": self.makam,
+                "tonic": self.tonic,
+                "mbid": self.mbid,
+                "duration": self.mb_tags["duration"],
+                "metadata": self.mb_tags,
+            },
+        )
+
+
+@io.coerce_to_string_io
+def load_pitch(fhandle: TextIO) -> annotations.F0Data:
+    """Load pitch
+
+    Args:
+        pitch path (str): Local path where the pitch annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        F0Data: pitch annotation
+
+    """
+    time_step = 0.0029
+
+    times = []
+    freqs = []
+    reader = csv.reader(fhandle, delimiter=",")
+    for line in reader:
+        freqs.append(float(line[0]))
+
+    for i in np.arange(len(freqs)):
+        times.append(round(float(time_step * i), 4))
```

Comment:
Just to double check, can you confirm that 1) the hop size is exactly 0.0029 s, and that the frames are left aligned, and not center aligned? In the mirdata paper, this was one of the big sources of differences when evaluating on ikala - slight differences in the hop size and in the alignment of the time stamps

cc @nkundiushuti

Reply:
Yes! The dataset's README states the following : _the timestamps are removed to reduce the filesizes. The step size of the pitch track is 0.0029 seconds (an analysis window of 128 sample hop size of an mp3 with 44100 Hz sample rate), with which one can recompute the timestamps of samples._ And so I did!

Reply:
Ah great! if that's the case, can we use `time_step = 128 / 44100`? It's very close to 0.0029, but over time the small difference propagates! And we can remove the `round`.

On the frame alignment, it looks like it's not clear from the documentation. Maybe we should check it with the dataset creators? Center aligned frames are more likely to be the "correct" way to align it, but maybe not!

Reply:
💯 for `time_step = 128 / 44100 seconds`, 0.0029 is not clear.

f0 was computed using [this tool](https://github.com/sertansenturk/predominantmelodymakam/tree/f8b7302bc657f90e2b10a0ffd988902935adc3d6), which basically wraps Essentia. In essence, we should check how [FrameGenerator](https://essentia.upf.edu/reference/std_FrameGenerator.html) dealt with alignment.

The current doc specifies centering as:

```
startFromZero (bool ∈ {true, false}, default = false) :
   whether to start the first frame at time 0 (centered at frameSize/2) if true, or -frameSize/2 otherwise (zero-centered)
```

However, we may want to ensure that the algorithm worked the same circa 2016. 
We seem to have noted the Essentia version as ([commit: e2339c](https://github.com/MTG/essentia/commit/e2339c11f869a850fc15a3fff7a75264db92fe11)) in this [JSON file](https://github.com/MTG/otmm_makam_recognition_dataset/blob/dlfm2016/algorithms.json).

@genisplaja if you can confirm the answer from Essentia's source code by yourself, I'd be glad. Otherwise, I'll dig more over the weekend.

Reply:
Thanks @rabitt for the suggestion, I'll fix it right now.

Sure @sertansenturk, thank you very much for all the information! I've gone through both implementations and it looks like the algorithm hasn't changed at all, and the default parameters are configured the same, so the frames should be zero-centered!